pom.xml

```diff
@@ -2355,6 +2355,10 @@ under the License.
 								<exclude>@org.apache.flink.annotation.Experimental</exclude>
 								<exclude>@org.apache.flink.annotation.Internal</exclude>
 								<!-- MARKER: start exclusions; these will be wiped by tools/releasing/update_japicmp_configuration.sh -->
+								<!-- New method has been added to the interface, but this interface shouldn't be implemented by users, only used/called, so this doesn't break API compatibility. -->
+								<exclude>org.apache.flink.api.common.eventtime.WatermarkGeneratorSupplier$Context</exclude>
+								<!-- Base interface has been extracted from the existing Clock class. This interface shouldn't affect user code. -->
+								<exclude>org.apache.flink.util.clock.Clock</exclude>
```

Comment:
Please pay attention to this

Reply:
It would be nice to return System clock from the new method (`WatermarkGeneratorSupplier.Context#getInputActivityClock`) to avoid breaking existing implementations.
I think it's not impossible that there are custom implementations of this interface.

Reply:
This context is provided by Flink to operators/functions, so I don't see any legitimate way how user should be interested in implementing it's own context. This could only happen if someone forks Flink and extends it somehow in this area AFAIU. I will merge it as is.

Reply:
Actually this `WatermarkGeneratorSupplier.Context#getInputActivityClock`, turns out to be a problem and user code actually implements on its own `WatermarkGeneratorSupplier.Context` interface. For example our Kafka connector does this here: https://github.com/apache/flink-connector-kafka/blob/main/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/internals/AbstractFetcher.java#L438-L439

Due to this reason, I don't see an easy way how to backport this bug fix to release 1.19 and 1.20 and I would suggest to only fix the bug in 2.0. Especially given that there doesn't seem to be a large interest in this bug fix (it hasn't been reported before). I will revert my changes for 1.19 and 1.20